plugins/eos-blacklist/gs-plugin-eos-blacklist.c

```diff
@@ -238,19 +238,18 @@ static gboolean
 gs_plugin_locale_is_compatible (GsPlugin *plugin,
 				const char *locale)
 {
-	g_auto(GStrv) locale_variants;
+	g_auto(GStrv) locale_variants = NULL;
 	g_auto(GStrv) locale_options = NULL;
-	const char *plugin_locale = gs_plugin_get_locale (plugin);
-
-        /* Check if a variant of a locale is compatible, e.g. 'pt' for 'pt_BR' */
-	locale_variants = g_get_locale_variants (plugin_locale);
-	if (locale_variants != NULL && g_strv_contains ((const gchar * const *) locale_variants, locale))
-		return TRUE;
+	int idx;
 
         /* check if the app's locale is listed in the languages key for the ostree repo file */
 	locale_options = get_default_locales (plugin);
-	if (locale_options != NULL && g_strv_contains ((const gchar * const *) locale_options, locale))
-		return TRUE;
+	for (idx = 0; locale_options[idx] != NULL; idx++) {
+		/* Check if a variant of a locale is compatible, e.g. 'pt' for 'pt_BR' */
+		locale_variants = g_get_locale_variants (locale_options[idx]);
```

Comment:
This is the wrong way round. We know that the entries of `locale_options` will always be two-character language codes, like `pt`. It is the app's suffix, stored in `locale`, which may be either `pt_BR` or `pt` (for example).

Reply:
Sorry, this edit was just so I could save my work to switch vms haha